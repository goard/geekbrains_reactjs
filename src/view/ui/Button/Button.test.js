import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom';

import ButtonCustom from '.';

describe('ButtonCustom', () => {
  it('render component Button', () => {
    render(<ButtonCustom>Change name</ButtonCustom>);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<ButtonCustom>Change name</ButtonCustom>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text name button', () => {
    render(<ButtonCustom>Change name</ButtonCustom>);
    expect(screen.getByText(/Change name/)).toBeInTheDocument();
  });

  it('render multiply component', () => {
    render(
      <>
        <ButtonCustom>Edet</ButtonCustom>
        <ButtonCustom>Delete</ButtonCustom>
      </>
    );
    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button is disable', () => {
    render(<ButtonCustom disabled>Delete</ButtonCustom>);
    expect(screen.getByText(/Delete/)).toBeDisabled();
  });

  it('button have style color green', () => {
    render(<ButtonCustom>Delete</ButtonCustom>);
    expect(screen.getByText(/Delete/)).toHaveStyle({ color: 'green' });
  });

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn();

    render(<ButtonCustom click={mockHandler}>Delete</ButtonCustom>);
    await userEvent.click(screen.getByText(/Delete/));
    expect(mockHandler).toHaveBeenCalledTimes(1);
  });
});
