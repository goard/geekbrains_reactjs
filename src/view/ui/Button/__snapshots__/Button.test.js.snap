// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCustom render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    style="color: green;"
    tabindex="0"
    type="button"
  >
    Change name
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
